src/ensenso.cpp

```diff
@@ -33,6 +25,19 @@ NxLibInitGuard & NxLibInitGuard::operator=(NxLibInitGuard && other) {
 	return *this;
 }
 
+Result<NxLibInitToken> NxLibInitGuard::initNxLib() {
+	int error = 0;
+	nxLibInitialize(error);
+	if (error) {
+		// TODO: check if this is the correct error!!
+		Result<NxCommandError> nx_error = NxCommandError::getCurrent();
+		if (!nx_error) return estd::error("failed to initialize nx library.");
```

Comment:
```suggestion
		if (!nx_error) return estd::error("failed to initialize nx library");
```
Error messages should be composable :)